packages/server/src/lib/history.ts

```diff
@@ -80,25 +80,14 @@ export const historyWithinArticlesViewedSettings = (
         return true;
     }
 
-    const { minViews, maxViews, periodInWeeks } = articlesViewedSettings;
+    const { minViews, maxViews, periodInWeeks, tagId } = articlesViewedSettings;
+
+    const viewCountForWeeks = tagId
+        ? getArticleViewCountByTagForWeeks(tagId, history, periodInWeeks, now)
+        : getArticleViewCountForWeeks(history, periodInWeeks, now);
 
-    const viewCountForWeeks = getArticleViewCountForWeeks(history, periodInWeeks, now);
     const minViewsOk = minViews ? viewCountForWeeks >= minViews : true;
     const maxViewsOk = maxViews ? viewCountForWeeks <= maxViews : true;
 
     return minViewsOk && maxViewsOk;
 };
-
-export const historyWithinArticlesViewedSettingsByTag = (
```

Comment:
the `historyWithinArticlesViewedSettings` function now does any filtering by tag